apps/src/templates/rubrics/LearningGoals.jsx

```diff
@@ -176,6 +177,9 @@ export default function LearningGoals({
   const [currentLearningGoal, setCurrentLearningGoal] = useState(0);
   const learningGoalEvalIds = useRef(Array(learningGoals.length).fill(null));
   const teacherFeedbacks = useRef(Array(learningGoals.length).fill(''));
+  const teacherFeedbacksLoaded = useRef(
+    Array(learningGoals.length).fill(false)
+  );
```

Comment:
@wilkie  I think you called this a shallow ref, is that right and if so what does that mean? it's not jumping to the top of my search results

Reply:
Oh, this is just representing that it had loaded the saved feedback for a particular learning goal. That's a copy from the similar ref storing the actual saved feedback.

The `useRef` creating a ref here is useful because you can set a ref without causing a re-render of a component. That's good because we load these all in the background and only selectively show the current one. So there's logic in the fetch that will put this to `true` for the learning goal and then logic there to, if it is the current index, set the `loaded` state to true.

I said it is 'shallow' because the variable doesn't not represent the value itself... you have to query it via `.current`. So when you pass around the variable (or trap it inside a Promise), that value is still negotiated when it is needed.

When you, say, hit the `index === currentLearningGoal` inside the fetch, which I said is a problem, the value that `currentLearningGoal` represents is trapped inside the fetch... frozen at that value when the fetch originally happened. When the fetch finishes, `setCurrentLearningGoal` might have been called, but it won't know that and it will be using the old value. So it will update the screen.

Since it freezes variables inside the Promise, we need a shallow representation that freezes only the function to retrieve the value instead of the value itself. That's what `useRef` is for... but since references cannot re-render components... and we want to re-render when the current learning goal changes, we annoyingly seem to need both a ref and a state to represent the same thing. But with the ref in place, the `then` clauses of the `fetch` will query the value at the current moment in time, and appropriate update the display.

Reply:
To be clear `currentLearningGoalRef` was what I referred to colloquially as a 'shallow ref'.

Reply:
Even though, hmm, all refs are technically shallow haha.